File: apiEZPZ/src/main/java/com/example/apiezpz/auth/repository/UserRepository.java
package com.example.apiezpz.auth.repository;

import com.example.apiezpz.auth.entity.User;
import org.springframework.data.jpa.repository.JpaRepository;

public interface UserRepository extends JpaRepository<User, Long> {
    boolean existsByUsername(String username);
    User findByUsername(String username);
}

File: apiEZPZ/src/main/kotlin/com/example/apiezpz/auth/repository/UserRepository.kt
package com.example.apiezpz.auth.repository

import com.example.apiezpz.auth.entity.User
import org.springframework.data.jpa.repository.JpaRepository

interface UserRepository : JpaRepository<User, Long> {
    fun existsByUsername(username: String): Boolean
    fun findByUsername(username: String): User?
} 